tests/dialects/test_teradata.py

```diff
@@ -210,3 +210,73 @@ def test_cast(self):
                 "teradata": "TRYCAST('-2.5' AS DECIMAL(5, 2))",
             },
         )
+
+    def test_time(self):
+        self.validate_identity("SELECT CURRENT_TIMESTAMP")
+
+        self.validate_all(
+            "CURRENT_TIMESTAMP",
+            read={
+                "snowflake": "CURRENT_TIMESTAMP()",
+            },
+            write={
+                "teradata": "CURRENT_TIMESTAMP",
+                "snowflake": "CURRENT_TIMESTAMP()",
+            },
+        )
+
+        self.validate_all(
+            "SELECT DATE_ADD('2023-01-01', 5, YEAR)",
+            write={
+                "snowflake": "SELECT DATEADD(YEAR, 5, '2023-01-01')",
+                "teradata": "SELECT '2023-01-01' + INTERVAL '5' YEAR",
+            },
+        )
+        self.validate_all(
+            "SELECT DATE_ADD('2023-01-01', -5, YEAR)",
+            write={
+                "snowflake": "SELECT DATEADD(YEAR, -5, '2023-01-01')",
+                "teradata": "SELECT '2023-01-01' - INTERVAL '5' YEAR",
+            },
+        )
+        self.validate_all(
+            "SELECT DATE_SUB('2023-01-01', 5, YEAR)",
+            write={
+                "teradata": "SELECT '2023-01-01' - INTERVAL '5' YEAR",
+            },
+        )
+        self.validate_all(
+            "SELECT DATE_SUB('2023-01-01', -5, YEAR)",
+            write={
+                "teradata": "SELECT '2023-01-01' + INTERVAL '5' YEAR",
+            },
+        )
+        self.validate_all(
+            "SELECT INTERVAL '1 quarter'",
+            write={"teradata": "SELECT (90 * INTERVAL '1' DAY)"},
+        )
+        self.validate_all(
+            "SELECT INTERVAL '1 week'",
+            write={"teradata": "SELECT (7 * INTERVAL '1' DAY)"},
+        )
```

Comment:
Ditto about using `self.validate_all` with `read` here. In general, though, when testing X -> X we prefer to use `self.validate_identity` because it's more compact.

Reply:
Makes sense. I think in this case I'll make the sql the teradata sql, and then read the 1 week expression to make sure that we translate weeks to days